d2compiler/compile.go

```diff
@@ -189,7 +189,7 @@ func (c *compiler) compileField(obj *d2graph.Object, f *d2ir.Field) {
 			}
 		}
 		scopeObjIDA := d2ir.IDA(fr.Context.ScopeMap)
-		scopeObj, _ := obj.Graph.Root.HasChildIDVal(scopeObjIDA)
+		scopeObj := obj.Graph.Root.EnsureChildIDVal(scopeObjIDA)
```

Comment:
How did this fix the problem? Wouldn't the scopeObj be guaranteed to exist?

Reply:
i msged u in slack:  "i think cuz u compile depth-first, the outer object referenced by an underscore key may not exist yet, so scopeobj is nil when compiling its references"

Reply:
> i msged u in slack: "i think cuz u compile depth-first, the outer object referenced by an underscore key may not exist yet, so scopeobj is nil when compiling its references"

That doesn't make sense to me. We create all parent objects with `obj = obj.EnsureChild(d2graphIDA([]string{f.Name}))` first on line 174.